carma-messenger-core/cpp_message/test/test_EmergencyVehicleAck.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2019-2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "cpp_message/EmergencyVehicleAck_Message.h"
+#include <gtest/gtest.h>
+#include <boost/optional/optional_io.hpp> //to print boost::optional
+
+TEST(EmergencyVehicleAckMessageTest, testDecodeEmergencyVehicleAckMsg)
+{
+    // binary copied from the output of testEncodeEmergencyVehicleAckMsg below
+    
+    std::vector<uint8_t> binary_input = {0, 247, 76, 77, 90, 113, 39, 212, 90, 209, 171, 22, 12, 38, 173, 56, 147, 234, 45, 104, 213, 131, 150, 172, 88, 65, 133, 14, 36, 88, 204, 88, 177, 98, 197, 139, 22, 43, 89, 50, 100, 201, 107, 54, 108, 217, 173, 131, 6, 12, 21, 172, 88, 177, 98, 197, 139, 22, 44, 88, 177, 98, 229, 147, 38, 108, 219, 178, 96, 205, 179, 134, 173, 27, 183, 106, 225, 131, 112};
+    auto node = std::make_shared<rclcpp::Node>("test_node");
+    cpp_message::Emergency_Vehicle_Ack worker(node->get_node_logging_interface());
+    auto res = worker.decode_emergency_vehicle_ack_message(binary_input);
+    carma_v2x_msgs::msg::EmergencyVehicleAck to_read;
+    if (res)
+    {
+        to_read = res.get();
+        if (to_read.m_header.plan_id == "11111111-2222-3333-AAAA-111111111111" && to_read.acknowledgement == 0)
+        {
+            EXPECT_TRUE(true);
+        }
+        else
+            EXPECT_TRUE(false);
+    }
+    else
+        EXPECT_TRUE(false);
+}
+
+TEST(EmergencyVehicleAckMessageTest, testEncodeEmergencyVehicleAckMsg)
+{
+    auto node = std::make_shared<rclcpp::Node>("test_node");
+    cpp_message::Emergency_Vehicle_Ack worker(node->get_node_logging_interface());
+    carma_v2x_msgs::msg::MobilityHeader header;
+    carma_v2x_msgs::msg::EmergencyVehicleAck message;
+    header.sender_id = "USDOT-45100";
+    header.recipient_id = "USDOT-45095";
+    header.sender_bsm_id = "10ABCDEF";
+    header.plan_id = "11111111-2222-3333-AAAA-111111111111";
+    header.timestamp = 9223372036854775807;
+    message.m_header = header;
+    message.acknowledgement = 0;
+    auto res = worker.encode_emergency_vehicle_ack_message(message);
+
+    if (res)
+    {
+        // std::vector<uint8_t> to_read=res.get();
```

Comment:
Can we remove this comment block? Otherwise, if it should be left in to help someone running future unit tests on this node, let's add a comment at the top of it explaining why it is useful